kubernetes/structures_container.go

```diff
@@ -348,15 +348,11 @@ func flattenContainerPorts(in []v1.ContainerPort) []interface{} {
 	return att
 }
 
-func flattenContainerResourceRequirements(in v1.ResourceRequirements) ([]interface{}, error) {
+func flattenContainerResourceRequirements(in v1.ResourceRequirements) []interface{} {
 	att := make(map[string]interface{})
-	if len(in.Limits) > 0 {
-		att["limits"] = flattenResourceList(in.Limits)
-	}
-	if len(in.Requests) > 0 {
-		att["requests"] = flattenResourceList(in.Requests)
-	}
-	return []interface{}{att}, nil
```

Comment:
Remove length validation here and accept an empty map from `flattenResourceList` to store empty values instead of `null` in the Terraform state file.